spec/datadog/appsec/contrib/http/faraday_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require 'datadog/appsec/spec_helper'
+require 'faraday'
+
+RSpec.describe 'AppSec Faraday integration' do
+  let(:telemetry) { instance_double(Datadog::Core::Telemetry::Component) }
+  let(:ruleset) { Datadog::AppSec::Processor::RuleLoader.load_rules(ruleset: :recommended, telemetry: telemetry) }
+  let(:processor) { Datadog::AppSec::Processor.new(ruleset: ruleset, telemetry: telemetry) }
+  let(:context) { Datadog::AppSec::Context.new(trace, span, processor) }
+
+  let(:span) { Datadog::Tracing::SpanOperation.new('root') }
+  let(:trace) { Datadog::Tracing::TraceOperation.new }
+
+  let(:client) do
+    ::Faraday.new('http://example.com') do |faraday|
+      faraday.adapter(:test) do |stub|
+        stub.get('/success') { |_| [200, {}, 'OK'] }
+      end
+    end
+  end
+
+  before do
+    Datadog.configure do |c|
+      c.appsec.enabled = true
+      c.appsec.instrument :faraday
+    end
+
+    Datadog::AppSec::Context.activate(context)
+  end
+
+  after do
+    Datadog.configuration.reset!
+
+    Datadog::AppSec::Context.deactivate
+    processor.finalize
+  end
+
+  context 'when RASP is disabled' do
+    before do
+      allow(Datadog::AppSec).to receive(:rasp_enabled?).and_return(false)
+    end
+
+    it 'does not call waf when making a request' do
+      expect(Datadog::AppSec.active_context).not_to receive(:run_rasp)
+
+      client.get('/success')
+    end
+
+    it 'returns the http response' do
+      response = client.get('/success')
+
+      expect(response.status).to eq(200)
+      expect(response.body).to eq('OK')
+    end
+  end
+
+  context 'when there is no active context' do
+    before do
+      Datadog::AppSec::Context.deactivate
+    end
+
+    it 'does not call waf when making a request' do
+      expect(Datadog::AppSec.active_context).not_to receive(:run_rasp)
+
+      client.get('/success')
+    end
+
+    it 'returns the http response' do
+      response = client.get('/success')
+
+      expect(response.status).to eq(200)
+      expect(response.body).to eq('OK')
+    end
+  end
+
+  context 'when RASP is enabled' do
+    before do
+      allow(Datadog::AppSec).to receive(:rasp_enabled?).and_return(true)
+    end
+
+    it 'calls waf with correct arguments when making a request' do
+      expect(Datadog::AppSec.active_context).to(
+        receive(:run_rasp).with(
+          Datadog::AppSec::Ext::RASP_SSRF,
+          {},
+          { 'server.io.net.url' => 'http://example.com/success' },
+          Datadog.configuration.appsec.waf_timeout
+        ).and_call_original
```

Comment:
I have that feeling that it looks like an integration test, but instead of behavior testing we are doing a mix of unit-testing here, it will be good to do just a single way.

This explain why do we need to configure rules, because if we do unit-test we should not, but if we do integration-test - we do.

Reply:
I think the goal of this test is to ensure that we are patching faraday correctly (different versions of the gem). I think that it is enough to test that we are calling `run_rasp` with correct arguments?

Reply:
Yes, but why call original then? We can stop at the RASP check, right?

Reply:
yes, you're right, it's already part of the integration test. And if we don't run RASP, we can also work with a context stub here.

Reply:
changed specs in this commit [b7fbdd1](https://github.com/DataDog/dd-trace-rb/pull/4391/commits/b7fbdd1eb6a0e894e1e10c8315c406f42fb22d29)